### PR TITLE
⚠️ Update versions of dependencies (go to 1.16 and controller runtime to v0.8.3)

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Update versions of dependencies:
- go lang to 1.16
- controller runtime to v0.8.3
- cluster api to latest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
update version of  go lang to 1.16
update version of controller runtime to v0.8.3
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
needs test infra PR to updates images to go 1.16: https://github.com/kubernetes/test-infra/pull/21273

**Comment by nader-ziada** [comment]
/hold

for https://github.com/kubernetes/test-infra/pull/21273

File: templates/cluster-template-ephemeral.yaml [hunk 1]
```diff
@@ -169,11 +167,9 @@ spec:
     spec:
       location: ${AZURE_LOCATION}
       osDisk:
-        diffDiskSettings:
-          option: Local
         diskSizeGB: 128
         managedDisk:
-          storageAccountType: Standard_LRS
+          storageAccountType: Premium_LRS
```
Comment (devigned) [inline templates/cluster-template-ephemeral.yaml:172]: This seems unrelated updating to 1.16. Was this needed or is related to another change set?

File: templates/test/cluster-template-prow-nvidia-gpu.yaml [hunk 2]
```diff
@@ -175,10 +175,10 @@ spec:
       osDisk:
         diskSizeGB: 128
         managedDisk:
-          storageAccountType: ${AZURE_GPU_NODE_STORAGE_TYPE:=Standard_LRS}
+          storageAccountType: Premium_LRS
         osType: Linux
       sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
-      vmSize: ${AZURE_GPU_NODE_MACHINE_TYPE:=Standard_NV6}
+      vmSize: ${AZURE_NODE_MACHINE_TYPE}
```
Comment (devigned) [inline templates/test/cluster-template-prow-nvidia-gpu.yaml:181]: This seems unrelated updating to 1.16. Was this needed or is related to another change set?

**Review by devigned — COMMENTED** [review]
Looks good. Just a couple questions about template changes.

File: templates/cluster-template-ephemeral.yaml (hunk 1 above)
Comment (nader-ziada) [reply templates/cluster-template-ephemeral.yaml:172]: I think it happened somehow from the generate, i'll look into it

Comment (nader-ziada) [reply templates/cluster-template-ephemeral.yaml:172]: I might have messed up a rebase 🤷

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Review by devigned — APPROVED** [review]
/lgtm
/approve
